--- libsvg-0.1.4/configure.in.orig	2005-06-15 22:27:49.000000000 +0200
+++ libsvg-0.1.4/configure.in	2005-06-19 10:48:26.551088024 +0200
@@ -55,6 +55,12 @@
   LIBSVG_PARSER_BACKEND="libxml"
   LIBSVG_REQUIRES="libxml-2.0"
 fi
+
+AC_CHECK_LIB(jpeg, jpeg_start_decompress, [LIBSVG_LIBS="$LIBSVG_LIBS -ljpeg"],
+	[AC_MSG_ERROR([libjpeg is required])])
+AC_CHECK_LIB(png12, png_create_read_struct, [LIBSVG_LIBS="$LIBSVG_LIBS -lpng12"],
+	[AC_MSG_ERROR([libpng is required])])
+
 AC_SUBST(LIBSVG_CFLAGS)
 AC_SUBST(LIBSVG_LIBS)
 AC_SUBST(LIBSVG_REQUIRES)
