diff -ur libsvg-0.1.4/configure.in libsvg-0.1.4-lt/configure.in
--- libsvg-0.1.4/configure.in	2009-05-13 16:10:01.000000000 +0200
+++ libsvg-0.1.4-lt/configure.in	2009-05-13 16:09:26.000000000 +0200
@@ -8,18 +8,18 @@
 # libtool shared library version
 
 # Increment if the interface has additions, changes, removals.
-LT_CURRENT=1
+LTLIB_CURRENT=1
 
 # Increment any time the source changes; set to
 # 0 if you increment CURRENT
-LT_REVISION=0
+LTLIB_REVISION=0
 
 # Increment if any interfaces have been added; set to 0
 # if any interfaces have been removed. removal has
 # precedence over adding, so set to 0 if both happened.
-LT_AGE=0
+LTLIB_AGE=0
 
-VERSION_INFO="$LT_CURRENT:$LT_REVISION:$LT_AGE"
+VERSION_INFO="$LTLIB_CURRENT:$LTLIB_REVISION:$LTLIB_AGE"
 AC_SUBST(VERSION_INFO)
 
 dnl ===========================================================================
